### PR TITLE
node Buffer can be constructed from ArrayBuffer and Uint8Array

### PR DESCRIPTION
**Pull request by philikon**
Also provide declaration for new Buffer.from() which along with Buffer.alloc() is to replace the constructor

### CODE REVIEW

File: lib/node.js [hunk 1]
```diff
@@ -18,10 +18,13 @@ type buffer$NonBufferEncoding =
   'utf16le' | 'UTF16LE' | 'utf-16le' | 'UTF-16LE';
 type buffer$Encoding = buffer$NonBufferEncoding | 'buffer'
 
+type buffer$Source = Array<number> | string | ArrayBuffer | Uint8Array;
```
Comment (mroch) [inline lib/node.js]: both places this is used also do `| Buffer`. does it make sense to add `Buffer` to this union?

**Comment by mroch** [comment]
you'll need to re-record the node tests (or I can do it when I merge if you want)

```
./runtests.sh bin/flow node_tests  # will fail
cp tests/node_tests/*.{out,exp}
./runtests.sh bin/flow node_tests  # will pass now
```

**Comment by mroch** [comment]
just also saw https://github.com/facebook/flow/pull/1978 which adds Buffer.from but not ArrayBuffer/Uint8Array. I'm going to take that one first

File: lib/node.js (hunk 1 above)
Comment (philikon) [inline lib/node.js]: I guess. I didn't know whether I could reference `Buffer` before it's declaration...

**Comment by philikon** [comment]
> just also saw #1978 which adds Buffer.from but not ArrayBuffer/Uint8Array. I'm going to take that one first

Ah, fair point. I'll rebase and see if I still need to do anything here.

**Comment by facebook-github-bot-5** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Pull request closed without merging**